lib/galaxy/tools/cache.py

```diff
@@ -53,7 +53,7 @@ def _get_cache(self, flag='r', create_if_necessary=False):
                 if flag == 'r':
                     self._cache.flag = flag
             else:
-                self._cache = SqliteDict(self.cache_file, flag=flag, encode=encoder, decode=decoder, autocommit=False)
+                self._cache = SqliteDict(self.writeable_cache_file and self.writeable_cache_file.name or self.cache_file, flag=flag, encode=encoder, decode=decoder, autocommit=False)
```

Comment:
Can you split this in 2 statements for clarity?
Also the `... and ... or ` is a bit "Perlish", `self.writeable_cache_file.name if self.writeable_cache_file else self.cache_file` is more understandable?